ast/format_test.go

```diff
@@ -0,0 +1,222 @@
+package ast_test
+
+import (
+	"testing"
+
+	"github.com/influxdata/flux/ast"
+	"github.com/influxdata/flux/parser"
+	"github.com/pkg/errors"
+)
+
+var skip = map[string]string{
+	"array_expr":           "without pars -> bad syntax, with pars formatting removes them",
+	"conditional":          "how is a conditional expression defined in spec?",
+	"duration_multiple":    "how are multiple duration values meant to be used?",
+	"nil_value_as_default": "cannot distinguish nil values from default values in function definition",
```

Comment:
Why not? A `nil` should mean no default value. The `[]` should not be parsing as `nil`. There was a bug about this recently maybe you need to rebase your branch off master to get the fix?

Reply:
After rebase it works, thank you